infrastructure/network/network_service.go

```diff
@@ -80,14 +90,15 @@ func (n *networkService) IfaceCreate(ctx context.Context, input ports.IfaceCreat
 		return nil, fmt.Errorf("creating interface %s using netlink: %w", link.Attrs().Name, err)
 	}
 
-	macIf, err := netlink.LinkByName(link.Attrs().Name)
+	macIf, err := netlink.LinkByName(input.DeviceName)
 	if err != nil {
 		return nil, fmt.Errorf("getting interface %s using netlink: %w", link.Attrs().Name, err)
```

Comment:
is this error still correct? `input.DeviceName` now?

Reply:
I changed it back as there was no need to change that part. And with the original its more obvious that we are re-getting the link we just added.